pkg/resourcecollector/resourcecollector.go

```diff
@@ -755,6 +755,9 @@ func (r *ResourceCollector) pruneOwnedResources(
 							objectType.GetKind() != "StatefulSet" &&
 							objectType.GetKind() != "ReplicaSet" &&
 							objectType.GetKind() != "DeploymentConfig" &&
+							objectType.GetKind() != "ServiceAccount" &&
+							objectType.GetKind() != "Role" &&
+							objectType.GetKind() != "RoleBinding" &&
```

Comment:
Why do all these checks have to be in the for loop? We should check it outside of  `for _, owner := range owners `loop. Am I missing anything?

Reply:
There is a check for `if a pod is an owner` is there before we are checking for if object is not one among these types (deploy, sts, service etc...). If we take the decision of collecting the object before the for loop, there can be scenario the owner might be a pod itself and the real intention was to skip it.

Anyway we will get rid of all these checks and make it generic for all resources not to be collected  or something similar. So don't want to alter the logic now.